src/%IPM/ResourceProcessor/Default/Routine.cls

```diff
@@ -27,10 +27,29 @@ Method OnConfigureMappings(ByRef pParams) As %Status
 		Set tGlobalScope = ..ResourceReference.Module.GlobalScope && '$Get(pParams("Reload","ForceLocalScope"),0)
 		If 'tGlobalScope && '..LoadAsDirectory {
 			Set tSourceDB = ##class(%IPM.Utils.Module).GetRoutineDatabase($Namespace)
-			Set tSC = ##class(%IPM.Utils.Module).AddRoutineMapping($namespace,tName,,tSourceDB)
+			// Check if current file exists in the the routine database of the current namespace exists in current namespace('s database)
+			// If yes, skip creating unnecessary mappings
+			// (Spec,Dir=1,OrderBy=1,SystemFiles=1,Flat,NotStudio=0,ShowGenerated=0,Filter,RoundTime=0,Mapped=0)
+			// Set mapped=0 since we only want to check whether the file exists in the current routine DB
+			Set tResult = ##class(%Library.RoutineMgr).StudioOpenDialogFunc(tName_".*",1,1,1,,,,,,0)
```

Comment:
"tName.*"
The same piece of code, used multiple times, if we really need it, it has to be moved outside.

Reply:
Now moved to command helper class in %IPM.ResourceProcessor.Default.Document.cls!